app/javascript/controllers/user_partners_controller.js

```diff
@@ -0,0 +1,52 @@
+import { Controller } from "@hotwired/stimulus";
+
+export default class extends Controller {
+	static values = { permittedPartners: [String] };
+
+	connect() {
+		const getSelectValues = (select) => {
+			return [...(select && select.options)].reduce((accumulator, option) => {
+				if (option.selected) {
+					return [...accumulator, Number(option.value || option.text)];
+				}
+				return accumulator;
+			}, []);
+		};
+
+		$(this.element).select2();
+
+		$(this.element).on("select2:unselecting", (event) => {
+			const selectedValues = getSelectValues(this.element);
+			const permittedValues = this.permittedPartnersValue;
+			const selectedPermittedValues = permittedValues.filter((x) =>
+				selectedValues.includes(x)
+			);
+
+			if (
+				selectedPermittedValues.length <= 1 &&
+				permittedValues.includes(Number(event.params.args.data.id))
+			) {
+				if (
+					!confirm(
+						"Removing this partner will remove this user from your neighbourhood.\n\n Are you sure you want to remove it? \n\n  If you want to keep the user in the neighbourhood but still remove this partner from them, you'll need to make them an admin for another partner in your neighbourhood."
```

Comment:
```suggestion
"Removing this partner will remove this user from your neighbourhood.\n\nAre you sure you want to remove it? \n\nIf you want to keep the user in the neighbourhood but still remove this partner from them, you'll need to make them an admin for another partner in your neighbourhood."
```
minor formatting thing